hs_core/models.py

```diff
@@ -1354,6 +1303,77 @@ class AbstractResource(ResourcePermissionsMixin):
         class MyResourceContentType(pages.Page, hs_core.AbstractResource):
             ...
     """
+    # For each request for a resource page view, the Page will run can_view(), can_change(),
+    # can_delete() in succession to check for permissions, so for performance improvement,
+    # there might be merit to make ha_obj a class member variable with a persistent connection
+    # for the request, rather than a separate connection within each method call
+    def can_add(self, request):
+        return self.can_change(request)
+
+    def can_delete(self, request):
+        user = get_user(request)
+        if user.is_authenticated():
+            if user.is_superuser:
+                ret = True
+            elif self.creator and user.pk == self.creator.pk:
```

Comment:
We have to be careful of the semantics here.  There are two issues.

1.  Multiple ownership.  An owner can revoke their ownership which may result in them losing rights over the resource, but a resource is not to be left without an owner, so last remaining owner can not revoke ownership.  Also there is a primary owner (against who quota will eventually be counted) and their ownership can not be revoked.  Primary owner may be changed to another owner, then ownership of the first owner revoked (two steps).  I do not think the primary owner concept is implemented yet.

2.  Creator/author.  This is distinct from owner.  It records the intellectual contribution to the work (resource in this case).  The system defaults to filling in creator information from the user making the resource, who is also the owner.  But then creator/author can be changed without affecting rights at all.  Creator/author does not have to be a system user, can in fact even be deceased and may be an organization (that does not necessarily have any rights in the system).

The concept of original owner only plays a role in the history of the resource.  The user who first made the resource in the system is the original owner.  The system records and keeps a history of changes in ownership, as a matter of record, but being a past owner does not give any present rights.